acct_mgt_test.py

```diff
@@ -123,28 +123,35 @@ def compare_results(result, pattern):
     return False
 
 
-def oc_resource_exist(resource, kind, name, project=None):
+def oc_resource_exist(resource, kind, name, project=None) -> bool:
     """This uses oc to determine if an openshift resource exists"""
     result = None
-    if project is None:
-        result = subprocess.run(
-            ["oc", "-o", "json", "get", resource, name],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            check=False,
-        )
-    else:
-        result = subprocess.run(
-            ["oc", "-o", "json", "-n", project, "get", resource, name],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            check=False,
-        )
+    cmd = ["oc", "-o", "json"]
+    if project is not None:
+        cmd = cmd + ["-n", project]
+    cmd = cmd + ["get", resource]
+    if name is not None:
+        cmd = cmd + [name]
+
+    result = subprocess.run(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=False,
+    )
```

Comment:
ultimately we should just be using the openshift API, so spending time doing this may not be worthwhile.

Reply:
What I am attempting to do with oc is to use oc a reference for correct behavior.  This way, if an update changes the API, oc will have been changed to accommodate this before we get the change.  I am concerned if we are using the OpenShift Python API as our reference for the OpenShift Python API.

Reply:
I am not certain if currying the arguments (partial functions) would have benefits for long term maintenance.

Reply:
If we curried the parameters, wouldn't we have to pass in the partial function to any function that we want to use it in.  I just don't quite see how that would make things simpler.  For example:

    def stew(curried_func, resource, project_name=None):
        curried_func(["oc","get",resource])

where as the original call is just 2 parameters:
    
    def stew( resource, project_name=None)
          subprocess.run( ... )

and somewhere else you define the curried_func.

I worry about this approach as it would present a great opportunity for abuse, as well as increase the number of arguments in many of the functions.

An object oriented point of view is to realize that most of the calls have a similar argument list,
and, as such, can be made class methods.  The C++ statement is that a class function (or method) is
any function that implicitly takes the class as its first parameter.  I've though of attempting to simply things this way, but haven't yet decided on the approach to implement.

Reply:
you don't have to pass it in to a function. You'd define the partial function just like your other helper functions, and import it and use it wherever you need.

Let's say you have the partial function in `helper.py`
```python 
import subprocess
from functools import partial

command_runner = partial(subprocess.run, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=False)
```
and then in whatever module you need it, this is how we can use it:

``` python
from helper import command_runner

output = command_runner(["oc", "version"])
print(output)
```

But don't worry about it, I have approved this PR since this is not a blocker.

Reply:
another way would to define the helper function without using functools.partial would be:
```python3
import subprocess

def command_runner(command):
    return subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=False)
```
and then you just import and use command_runner everywhere instead of subprocess.

But again, this is only some optimization in my opinion and nothing critical.